io/src/main/scala/fs2/io/file/FileHandle.scala

```diff
@@ -111,9 +111,10 @@ private[file] object FileHandle {
         asyncCompletionHandler[F, Lock](f => chan.lock(position, size, shared, null, f))
 
       override def read(numBytes: Int, offset: Long): F[Option[Chunk[Byte]]] = {
-        val buf = ByteBuffer.allocate(numBytes)
```

Comment:
Wrapped these buffer allocations in `F.delay` so the returned action is safe to use more than once. Unrelated to memory leak.